### PR TITLE
Verify Proxy via Etherscan API

### LINKED ISSUE #565: verify proxy contracts via Etherscan
after code has been verified, can submit proxy address to Etherscan to verify the implementation address so people can interact with the contracts via etherscan
can do this programmatically via Etherscan API

### PR DESCRIPTION
**Pull request by anna-carroll**
closes #565

### CODE REVIEW

File: typescript/optics-deploy/src/verification/verifyProxy.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const axios = require("axios");
```
Comment (prestwich) [inline typescript/optics-deploy/src/verification/verifyProxy.ts:1]: `import`

File: typescript/optics-deploy/src/verification/verifyProxy.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export async function verifyProxy(network: string, address: string) {
+    // TODO: try other networks
+    // const prefix = network == 'mainnet' ? '' : `${network}.`;
```
Comment (prestwich) [inline typescript/optics-deploy/src/verification/verifyProxy.ts:5]: the pattern is `api` `api-kovan` `api-ropsten`

File: typescript/optics-deploy/src/verification/verifyProxy.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        throw new Error(verifyResponse.data.result);
+    }
+
+    // Check status of contract verification
+    const guid = verifyResponse.data.result;
+    const checkVerifyResponse = await axios.post("https://api.etherscan.io/api", null, {
+        params: {
+            module: "contract",
+            action: "checkproxyverification",
+            apikey: "IRCYZZTA5R8Y7MBK27DT49D3QMH7VZD7TD",
```
Comment (prestwich) [inline typescript/optics-deploy/src/verification/verifyProxy.ts:29]: need to rotate keys

Comment (anna-carroll) [reply typescript/optics-deploy/src/verification/verifyProxy.ts:29]: ah shit, i thought i didn't commit this, damn

Comment (anna-carroll) [reply typescript/optics-deploy/src/verification/verifyProxy.ts]: https://github.com/celo-org/optics-monorepo/pull/577/commits/eda4b02e46e5188fe70cfad562e627aa0dbec94c

File: typescript/optics-deploy/src/verification/verifyProxy.ts (hunk 2 above)
Comment (anna-carroll) [reply typescript/optics-deploy/src/verification/verifyProxy.ts]: https://github.com/celo-org/optics-monorepo/pull/577/commits/eda4b02e46e5188fe70cfad562e627aa0dbec94c

File: typescript/optics-deploy/src/verification/verifyProxy.ts (hunk 1 above)
Comment (anna-carroll) [reply typescript/optics-deploy/src/verification/verifyProxy.ts]: https://github.com/celo-org/optics-monorepo/pull/577/commits/eda4b02e46e5188fe70cfad562e627aa0dbec94c

**Comment by anna-carroll** [comment]
i cannot repro these tests failing locally 🤔 seems like i maybe need to install `axios` / `dotenv` in some other additional directory?

**Comment by prestwich** [comment]
okay here's your dependency bug

1. `optics-xapps` now imports `optics-deploy` in the `hardhatconfig`
2. `optics-deploy` relies on `dotenv` and `axios`
3. the `optics-xapps` github action doesn't install `optics-deploy`'s dependencies
4. therefore, when `optics-xapps` imports `optics-deploy`, `axios` is not installed in `optics-deploy`'s context

**Comment by prestwich** [comment]
solution:
I'm updating the `optics-xapps` workflow to install deps.

**Comment by prestwich** [comment]
drive-by changes:
1. remove dotenv from verification internals. change to passing the api key through 
2. run prettier
3. refactor hardhat configs in core and xapps to match eachother more closely
4. change requires to imports

**Review by prestwich — APPROVED** [review]

**Pull request merged**